pkg/webui/console/containers/device-onboarding-form/index.js

```diff
@@ -12,9 +12,75 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import DeviceOnboardingForm from './device-onboarding-form'
-import connect from './connect'
+import React, { useCallback } from 'react'
+import { merge } from 'lodash'
+import { useSelector } from 'react-redux'
 
-const ConnectedDeviceOnboardingForm = connect(DeviceOnboardingForm)
+import Form, { useFormContext } from '@ttn-lw/components/form'
+import SubmitBar from '@ttn-lw/components/submit-bar'
+import SubmitButton from '@ttn-lw/components/submit-button'
 
-export { ConnectedDeviceOnboardingForm as default, DeviceOnboardingForm }
+import sharedMessages from '@ttn-lw/lib/shared-messages'
+
+import { checkFromState } from '@account/lib/feature-checks'
+import { mayEditApplicationDeviceKeys } from '@console/lib/feature-checks'
+
+import DeviceProvisioningFormSection, {
+  initialValues as provisioningInitialValues,
+} from './device-provisioning-form-section'
+import DeviceTypeFormSection, {
+  initialValues as typeInitialValues,
+} from './device-type-form-section'
+import validationSchema from './validation-schema'
+
+const initialValues = merge({}, provisioningInitialValues, typeInitialValues)
+
+const DeviceOnboardingFormInner = () => {
+  const {
+    values: {
+      frequency_plan_id,
+      lorawan_version,
+      lorawan_phy_version,
+      ids: { join_eui },
+    },
+  } = useFormContext()
+  const maySubmit =
+    Boolean(frequency_plan_id) &&
+    Boolean(lorawan_version) &&
+    Boolean(lorawan_phy_version) &&
+    join_eui.length === 16
+
+  return (
+    <>
+      <DeviceTypeFormSection />
+      <DeviceProvisioningFormSection />
+      {maySubmit && (
+        <SubmitBar>
+          <Form.Submit message={sharedMessages.addDevice} component={SubmitButton} />
+        </SubmitBar>
+      )}
+    </>
+  )
+}
```

Comment:
I had to seperate the topmost form component into two components to be able to tap into the form context.
Alternatively one could have also done something like
```js
<Form
  ...
/>{({values}) => {
}}
</Form>
```
But I decided against that for consistency.